### PR TITLE
Set cuda version through a config param

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+options:
+  cuda-version:
+    type: string
+    default: "8.0.61"
+    description: |
+      The version of CUDA to be intalled.
```
Comment (tvansteenburgh) [inline config.yaml]: s/intalled/installed/

File: config.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  cuda-subversion:
```
Comment (tvansteenburgh) [inline config.yaml]: I think version and sub-version should be combined into one config param. There's no reason for them to be separate afaict - they are always used together.

**Review by tvansteenburgh — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 2 above)
Comment (ktsakalozos) [reply config.yaml]: There is only this place https://github.com/juju-solutions/layer-nvidia-cuda/blob/master/reactive/cuda.sh#L285 where CUDA_VERSION is used without CUDA_SUBVERSION. Should we use one config param and split on the last "-" to get the subversion?

Comment (ktsakalozos) [reply config.yaml]: CUDA_VERSION without the subversion is used also in install_nvidia_gdk (eg https://github.com/juju-solutions/layer-nvidia-cuda/blob/master/reactive/cuda.sh#L61) but I think this is dead code.

Comment (tvansteenburgh) [reply config.yaml]: Oh, I missed that one. But yes, let's split on the last '-'. I think that's better than having two config options.

Comment (ktsakalozos) [reply config.yaml]: Done!

File: reactive/cuda.sh [hunk 3]
```diff
@@ -3,8 +3,8 @@ set -ex
 
 source charms.reactive.sh
 
-CUDA_VERSION="8.0.61"
-CUDA_SUB_VERSION="1"
+CUDA_VERSION=$(config-get cuda-version | awk 'BEGIN{FS="-"}{print $1}')
+CUDA_SUB_VERSION=$(config-get cuda-version | awk 'BEGIN{FS="-"}{print $2}')
```
Comment (tvansteenburgh) [inline reactive/cuda.sh:7]: So we are assuming there will always be a sub version?

Comment (ktsakalozos) [reply reactive/cuda.sh:7]: If there is no subversion cuda.sh is completely wrong.

**Pull request merged**